NebulaAPI/Extensions/PlayerCollectionExtensions.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using NebulaAPI.GameState;
+using NebulaAPI.Networking;
+
+namespace NebulaAPI.Extensions;
+
+public static class PlayerCollectionExtensions
+{
+    public static IReadOnlyDictionary<INebulaConnection, INebulaPlayer> GetByConnectionStatus(
+        this IReadOnlyDictionary<INebulaConnection, INebulaPlayer> playerConnections, EConnectionStatus status)
+    {
+        return playerConnections
+            .Where(kvp => kvp.Key.ConnectionStatus == status)
+            .ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+    }
+
+    public static IReadOnlyCollection<INebulaPlayer> GetByConnectionStatus(
+        this IReadOnlyCollection<INebulaPlayer> players, EConnectionStatus status)
+    {
+        return players
+            .Where(value => value.Connection.ConnectionStatus == status)
+            .ToList();
+    }
+
+    public static INebulaPlayer GetByPlayerId(
+        this IReadOnlyCollection<INebulaPlayer> players, ushort id)
+    {
+        return players.FirstOrDefault(p => p.Id == id);
```

Comment:
this allocates a closure every time it's called.

Reply:
Any lambda in here that can't be marked static will likely allocate a closure, just a mention. That means that if it's called in hot code, it might trigger GCs.